src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -1472,4 +1472,18 @@ private static void EmitLdarg(ILGenerator emitter, int c)
             }
         }
     }
+
+    /// <summary>
+    /// The attribute for a PowerShell class to not affiliate with a particular Runspace\SessionState.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class NoRunspaceAffinityAttribute : ParsingBaseAttribute
```

Comment:
Maybe add "class" in the name to better discoverability? `NoRunspaceClassAffinityAttribute`

Reply:
> Maybe add "class" in the name to better discoverability? `NoRunspaceClassAffinityAttribute`

Tbh I don't think this is one that someone is gonna find on their own regardless. I don't think that's a problem we need to solve necessarily, but with that in mind I'd lean towards keeping the name in case we want to apply it to other things down the line.